docs/faq.md

```diff
@@ -1,21 +1,184 @@
+## Network
+
+### Can Fn do IO over HTTP or in general any TCP based communication?
+
+Like most functions platforms we support HTTP.
+
+### Can an Fn function connect to various endpoints to read/write data?
+
+Yes, a function is not restricted in what it can connect to.
+
+### Any limits on the network bandwidth?
+
+We currently don't yet offer Fn as a managed service which would manage network traffic.  Currently, there are no restrictions in the core Fn platform.
+
+<a id="Performance"></a>
+## Performance
+
+### How long can a function run? Can we make it run forever?
+
+We support 'hot functions' (see the end of [this tutorial](https://github.com/fnproject/tutorials/blob/master/JavaFDKIntroduction/README.md) for an example).  Hot functions will continue to live if they are used but, if not, will eventually be cleaned up.
+
+Function timeout is configurable. Please see <https://github.com/fnproject/fn/blob/master/docs/function-file.md>. Note though configurable, timeouts do have limits. For example, sync functions have a maximum upper limit of 120 seconds.
+
+### How does the service trace the liveliness of the function? If my function dies/crashes will the service provision it again?
+
+This is not a microservices platform so the notion is slightly different.  Functions are run when called.  If you call a function that doesn't exist then one will be started.
+
+### How does scaling work? Should this be initiated by a customer?
+
+Fn runs more instances of the function based on requests.  It is automatic. For more information see [scaling](operating/scaling.md).
+
+### When one host is overloaded, will this service automatically relocate the function to another host?
+
+When running in the cloud workloads are spread across available hosts.
+
+Functions currently do not relocate. Please see our [FnLB project inside Fn Project](https://github.com/fnproject/fn/tree/master/fnlb) for how to route requests to available hosts. Using the FnLB will relocate/scale functions if hosts are overloaded.
+
+<a id="Persistence"></a>
+## Persistence
+
+### Does your runtime provide any persistent store to save the function state information?
+
+State management is not part of Fn but you can use any storage service or database to store data.
+
+### Can we build a state function which will update itself based on the previous computation? Will it persist across function restarts?
+
+The problem you face is the lack of a guarantee of which instance of a hot function is called.  Standard practice is to externalize state.  The State Service is essentially Oracle Coherence which offers excellent performance as you may know.
 
-For now we require Docker primarily for the packaging and distribution via Docker Registries,
-but we've built Functions in a way that abstracts the container technology so we can support others as
-needed. For instance, we may add rkt support.
+Also check out Fn Flow if you’re writing Java (currently the primary supported language).
```

Comment:
I'd add a reason why. Something like "If your need for stateful functions is motivated by managing steps of a workflow that spans several functions (or several calls to the same function), check out Fn Flow..."

Reply:
Changed to this wording and added a link.